packages/walletconnect/src/index.ts

```diff
@@ -122,16 +126,11 @@ export class WalletConnect extends Connector {
     const cancelActivation = this.actions.startActivation()
 
     try {
-      await this.isomorphicInitialize()
-      if (!this.provider?.connected) throw Error('No existing connection')
-
-      // Wallets may resolve eth_chainId and hang on eth_accounts pending user interaction, which may include changing
-      // chains; they should be requested serially, with accounts first, so that the chainId can settle.
-      const accounts = await this.provider.request<string[]>({ method: 'eth_accounts' })
-      if (!accounts.length) throw new Error('No accounts returned')
-      const chainId = await this.provider.request<string>({ method: 'eth_chainId' })
-
-      this.actions.update({ chainId: parseChainId(chainId), accounts })
+      const { session, accounts, chainId } = await this.isomorphicInitialize()
```

Comment:
Can you test the case where you connect your wallet but specify a new chain? I'm afraid there might be a `Promise.all` type of thing hidden under this abstraction; see https://github.com/Uniswap/web3-react/pull/749.

Reply:
Let's pair on this tomorrow! It's a good call, but I am not sure how to trigger that state. I didn't see such possibility while connecting with WC2

> Serially requests accounts and then chainId, so that if a user changes chains whilst connecting

What happens right now is that once you want to connect your dapp with your wallet, you will see all the chains that dapp is requesting w/o ability to change that selection. Just accept/reject. 

I may have missed something, so feel free to try the preview link with Trust wallet and let me know!

Reply:
That sounds like WC has eliminated that bug at an architectural level :)